sklearn/metrics/tests/test_ranking.py

```diff
@@ -693,6 +696,107 @@ def test_score_scale_invariance():
     assert_equal(pr_auc, pr_auc_shifted)
 
 
+@pytest.mark.parametrize("y_true,y_score,expected_fpr,expected_fnr", [
+    ([0, 0, 1], [0, 0.5, 1], [0], [0]),
+    ([0, 0, 1], [0, 0.25, 0.5], [0], [0]),
+    ([0, 0, 1], [0.5, 0.75, 1], [0], [0]),
+    ([0, 0, 1], [0.25, 0.5, 0.75], [0], [0]),
+    ([0, 1, 0], [0, 0.5, 1], [0.5], [0]),
+    ([0, 1, 0], [0, 0.25, 0.5], [0.5], [0]),
+    ([0, 1, 0], [0.5, 0.75, 1], [0.5], [0]),
+    ([0, 1, 0], [0.25, 0.5, 0.75], [0.5], [0]),
+    ([0, 1, 1], [0, 0.5, 1], [0.0], [0]),
+    ([0, 1, 1], [0, 0.25, 0.5], [0], [0]),
+    ([0, 1, 1], [0.5, 0.75, 1], [0], [0]),
+    ([0, 1, 1], [0.25, 0.5, 0.75], [0], [0]),
+    ([1, 0, 0], [0, 0.5, 1], [1, 1, 0.5], [0, 1, 1]),
+    ([1, 0, 0], [0, 0.25, 0.5], [1, 1, 0.5], [0, 1, 1]),
+    ([1, 0, 0], [0.5, 0.75, 1], [1, 1, 0.5], [0, 1, 1]),
+    ([1, 0, 0], [0.25, 0.5, 0.75], [1, 1, 0.5], [0, 1, 1]),
+    ([1, 0, 1], [0, 0.5, 1], [1, 1, 0], [0, 0.5, 0.5]),
+    ([1, 0, 1], [0, 0.25, 0.5], [1, 1, 0], [0, 0.5, 0.5]),
+    ([1, 0, 1], [0.5, 0.75, 1], [1, 1, 0], [0, 0.5, 0.5]),
+    ([1, 0, 1], [0.25, 0.5, 0.75], [1, 1, 0], [0, 0.5, 0.5]),
+])
+def test_detection_error_tradeoff_curve_toydata(
+    y_true, y_score, expected_fpr, expected_fnr
+):
+    # Check on a batch of small examples.
+    fpr, fnr, _ = detection_error_tradeoff_curve(y_true, y_score)
+
+    assert_array_almost_equal(fpr, expected_fpr)
+    assert_array_almost_equal(fnr, expected_fnr)
+
+
+@pytest.mark.parametrize("y_true,y_score,expected_fpr,expected_fnr", [
+    ([1, 0], [0.5, 0.5], [1], [0]),
+    ([0, 1], [0.5, 0.5], [1], [0]),
+    ([0, 0, 1], [0.25, 0.5, 0.5], [0.5], [0]),
+    ([0, 1, 0], [0.25, 0.5, 0.5], [0.5], [0]),
+    ([0, 1, 1], [0.25, 0.5, 0.5], [0], [0]),
+    ([1, 0, 0], [0.25, 0.5, 0.5], [1], [0]),
+    ([1, 0, 1], [0.25, 0.5, 0.5], [1], [0]),
+    ([1, 1, 0], [0.25, 0.5, 0.5], [1], [0]),
+])
+def test_detection_error_tradeoff_curve_tie_handling(
+    y_true, y_score, expected_fpr, expected_fnr
+):
+    fpr, fnr, _ = detection_error_tradeoff_curve(y_true, y_score)
+
+    assert_array_almost_equal(fpr, expected_fpr)
+    assert_array_almost_equal(fnr, expected_fnr)
+
+
+def test_detection_error_tradeoff_curve_sanity_check():
+    # Exactly duplicated inputs yield the same result.
+    assert_array_almost_equal(
+        detection_error_tradeoff_curve([0, 0, 1], [0, 0.5, 1]),
+        detection_error_tradeoff_curve(
+            [0, 0, 0, 0, 1, 1], [0, 0, 0.5, 0.5, 1, 1])
+    )
+
+
+@pytest.mark.parametrize("y_score", [
+    (0), (0.25), (0.5), (0.75), (1)
+])
+def test_detection_error_tradeoff_curve_constant_scores(y_score):
+    # Check computation with perfect and constant scores.
```

Comment:
you're not testing with perfect scores, are you?

Reply:
Good catch, will add.